src/components/periodic-report/periodic-report.repository.ts

```diff
@@ -111,66 +112,89 @@ export class PeriodicReportRepository extends DtoRepository(IPeriodicReport) {
       );
   }
 
-  async reportForDate(
-    parentId: ID,
-    reportType: ReportType,
-    date: CalendarDate
-  ) {
-    return await this.db
+  async getCurrentDue(parentId: ID, reportType: ReportType) {
+    const res = await this.db
       .query()
       .match([
         node('baseNode', 'BaseNode', { id: parentId }),
         relation('out', '', 'report', { active: true }),
         node('node', `${reportType}Report`),
-      ])
-      .match([
-        node('node'),
-        relation('out', '', 'start', { active: true }),
-        node('start', 'Property'),
-      ])
-      .match([
-        node('node'),
         relation('out', '', 'end', { active: true }),
         node('end', 'Property'),
       ])
-      .raw(`WHERE start.value <= $date AND end.value >= $date`, {
-        date,
-      })
-      .return('node.id as id')
-      .asResult<{ id: ID }>()
+      .raw(`WHERE end.value < date()`)
+      .with('node, end')
+      .orderBy('end.value', 'desc')
+      .limit(1)
+      .apply(this.hydrate())
+      .return('dto')
+      .asResult<{ dto: UnsecuredDto<PeriodicReport> }>()
       .first();
+    return res?.dto;
   }
 
-  listEngagementReports(
-    engagementId: string,
-    { filter, ...input }: PeriodicReportListInput
-  ) {
-    return this.db
+  async getNextDue(parentId: ID, reportType: ReportType) {
+    const res = await this.db
       .query()
       .match([
-        node('engagement', 'Engagement', { id: engagementId }),
+        node('baseNode', 'BaseNode', { id: parentId }),
         relation('out', '', 'report', { active: true }),
-        node('node', 'ProgressReport'),
+        node('node', `${reportType}Report`),
+        relation('out', '', 'end', { active: true }),
+        node('end', 'Property'),
       ])
-      .apply(calculateTotalAndPaginateList(ProgressReport, input));
+      .raw(`WHERE end.value > date()`)
+      .with('node, end')
+      .orderBy('end.value', 'asc')
+      .limit(1)
+      .apply(this.hydrate())
+      .return('dto')
+      .asResult<{ dto: UnsecuredDto<PeriodicReport> }>()
+      .first();
+    return res?.dto;
   }
 
   async getLatestReportSubmitted(parentId: ID, type: ReportType) {
-    return await this.db
+    const res = await this.db
       .query()
       .match([
         node('', 'BaseNode', { id: parentId }),
         relation('out', '', 'report', { active: true }),
-        node('pr', `${type}Report`),
+        node('node', `${type}Report`),
         relation('out', '', 'start', { active: true }),
```

Comment:
For the sake of consistency I would use the report end to sort most recent. It doesn't really matter, but the other two queries do that.